File: chapter 08/recipe 8.2/DniScServerComponent/DniScServerComponentObj.cs
using System;
using System.Runtime.InteropServices;
using System.EnterpriseServices;

namespace DniScServerComponent
{
	[Description("Interface used to add numbers")]
	public interface IAddNumbers
	{
		int AddSomeNumbers(int numA, int numB);
	}

	[ClassInterface(ClassInterfaceType.None)]
	[Description("Add numbers server component")]
	public class DniScServerComponentObj
		: ServicedComponent, IAddNumbers
	{
		public int AddSomeNumbers(int numA, int numB)
		{
			return numA + numB;
		}
	}
}

File: chapter 08/recipe 8.2/DniScServerComponentVB/DniScServerComponentObj.vb
Imports System.Runtime.InteropServices
Imports System.EnterpriseServices

<Description("Interface used to add numbers")> _
Public Interface IAddNumbers
    Function AddSomeNumbers(ByVal numA As Integer, _
        ByVal numB As Integer) As Integer
End Interface

<ClassInterface(ClassInterfaceType.None)> _
<Description("Add numbers server component")> _
Public Class DniScServerComponentObj
    Inherits ServicedComponent
    Implements IAddNumbers

    Public Function AddSomeNumbers(ByVal numA As Integer, _
        ByVal numB As Integer) As Integer _
            Implements IAddNumbers.AddSomeNumbers

        Return numA + numB

    End Function
End Class
